txclib/utils.py

```diff
@@ -181,7 +186,9 @@ def make_request(method, host, url, username, password, fields=None,
             elif response.status == 404:
                 raise HttpNotFound(data)
             else:
-                raise Exception(data)
```

Comment:
@rigaspapas this case also covers the 400 Bad Request status code, which might have useful information about what went wrong in the response. If you want to handle ConnectionError separately you should check for error code >=500.